generate/genhcl/genhcl_test.go

```diff
@@ -1576,6 +1576,65 @@ func TestGenerateHCL(t *testing.T) {
 				},
 			},
 		},
+		{
+			name:  "generate HCL on stack with lets referencing indexing variable",
```

Comment:
the issue is not related to `lets` but I added a test for lets anyway because we will handle them the same soon.